### PR TITLE
Simplify Table binding to use ScriptBinding and leverage new SDK bind…

### PR DESCRIPTION
**Pull request by MikeStall**
…ing power.

Fix https://github.com/Azure/azure-webjobs-sdk-script/issues/753

This also causes C# and Node to share a codepath, and so fixes some C# cases that previously only worked in Node.

### CODE REVIEW

**Comment by MikeStall** [comment]
Note - I still need to update the nugget references to use the latest SDK. (but I need this https://github.com/Azure/azure-webjobs-sdk/pull/891 checked in first). So I tested the build here via direct copying the SDK dlls.

File: src/WebJobs.Script/Binding/ExtensionBinding.cs [hunk 1]
```diff
@@ -70,6 +71,12 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
                     context.Value = result;
                 }
             }
+            else if (_binding.DefaultType == typeof(JArray))
+            {
+                JArray entityArray = await context.Binder.BindAsync<JArray>(_attributes.ToArray());
+                string json = entityArray.ToString(Formatting.None);
```
Comment (mathewc) [inline src/WebJobs.Script/Binding/ExtensionBinding.cs]: Why is this conversion needed? I think later stages in the pipeline should convert as necessary. Note that we don't convert JObjects to string above.

File: src/WebJobs.Script/Binding/FunctionBinding.cs [hunk 2]
```diff
@@ -46,10 +46,6 @@ internal static Collection<FunctionBinding> GetBindings(ScriptHostConfiguration
                     string type = function.Type.ToLowerInvariant();
                     switch (type)
                     {
-                        case "table":
```
Comment (mathewc) [inline src/WebJobs.Script/Binding/FunctionBinding.cs]: 👍

File: src/WebJobs.Script/Binding/TableBinding.cs [hunk 3]
```diff
@@ -1,323 +0,0 @@
...
-namespace Microsoft.Azure.WebJobs.Script.Binding
-{
-    [CLSCompliant(false)]
-    public class TableBinding : FunctionBinding
-    {
-        private readonly BindingTemplate _partitionKeyBindingTemplate;
-        private readonly BindingTemplate _rowKeyBindingTemplate;
-        private readonly BindingTemplate _filterBindingTemplate;
-
-        public TableBinding(ScriptHostConfiguration config, TableBindingMetadata metadata, FileAccess access)
```
Comment (mathewc) [inline src/WebJobs.Script/Binding/TableBinding.cs]: We should be able to get rid of this TableBindingMetadata class as well, right?

File: src/WebJobs.Script/Binding/WebJobsCoreScriptBindingProvider.cs [hunk 4]
```diff
@@ -129,6 +133,80 @@ public override Collection<Attribute> GetAttributes()
...
 
+        private class TableScriptBinding : ScriptBinding
+        {
+            public TableScriptBinding(ScriptBindingContext context) : base(context)
+            {
+            }
+
+            private string TableName {
+                get { return Context.GetMetadataValue<string>("tableName"); }
+            }
```
Comment (mathewc) [inline src/WebJobs.Script/Binding/WebJobsCoreScriptBindingProvider.cs]: nit: fix spacing between blocks

File: src/WebJobs.Script/Binding/ExtensionBinding.cs (hunk 1 above)
Comment (MikeStall) [inline src/WebJobs.Script/Binding/ExtensionBinding.cs]: good catch. I was copying over code from the previous binding; but it looks like we don't need this. 

---
In reply to: [85890556](https://github.com/Azure/azure-webjobs-sdk-script/pull/858#discussion_r85890556) [](ancestors = 85890556)

**Comment by fabiocav** [comment]
@MikeStall looks like we have some reference issues:
https://ci.appveyor.com/project/appsvc/azure-webjobs-sdk-script-y8o14/build/1.0.10568-pcwivgxt/messages

**Comment by fabiocav** [comment]
Do you need to pull in updated SDK packages?

**Comment by MikeStall** [comment]
@fabiocav  - yes, I need updated SDK packages.  I tested this with a local frankenbuild.

**Comment by MikeStall** [comment]
CI is failing this test: Microsoft.Azure.WebJobs.Script.Tests.NodeEndToEndTests.TableInput . Need to investigate that before pushing this.

**Comment by mathewc** [comment]
After you pull your SDK change in to fix the last failing tests, :ship:

**Pull request merged**